src/ipc/ipc4/handler.c

```diff
@@ -537,7 +537,8 @@ static int ipc4_process_chain_dma(struct ipc4_message_request *ipc4)
 		if (ret)
 			tr_err(&ipc_tr, "failed to create chain dma %d", ret);
 
-		return ret;
+		if (!cdma.primary.r.enable)
```

Comment:
Shouldn't we return also if there was on error? e.g.
if (!cdma.primary.r.enable || ret)

Reply:
Sure. I agree.